### ISSUE TITLE
Общая структура сайта

### ISSUE THREAD

**Issue by Sergei-Litvinov** [tigran123/urantia-book-explorer#5]
Сейчас в начальном меню не предусмотрен резерв места для дальнейшего развития сайта, предусмотренного нашим ТЗ. Т.е. левое-боковое ниспадающее меню, в котором будут такие пункты как : "вторичные работы", "творчество читателей" и т.д.
Думаю надо сразу это учесть, чтобы потом не переделывать.

**Comment by tigran123** [comment]
Конечно, например Форума ещё нет, но я уже не только зарезервировал для него место, но и написал код, перехватывающий доступ к этому элементу и перенаправляющий весь сайт туда, где будет форум. Кстати, насчёт Форума я сегодня говорил с тех. поддержкой провайдера и они мне сказали, что таким образом как я это задумал сделать ничего не получится. Но мне кажется, что они просто ошибаются, ибо я не вижу технической причины, почему это невозможно. Но если они и окажутся правы --- не беда, придумаем как-нибудь иначе.

Что касается "вторичных работ", то надо помнить, что некоторые из них (например SRT комментарии о более чем 400-ах изменениях, сделанных в английском тексте Урантийских Документов) логически принадлежат в колонке "Заметки" ибо напрямую связаны со структурой текста. Некоторые другие материалы логически принадлежат секции "Ссылки/Links". Но для тех материалов, которые действительно заслуживают место на нашем сайте, разумеется будет добавлена соответствующая секция в меню --- места не жалко (всю панель меню ведь и отключить можно кнопкой Ctrl + H, если кому-то мешает работе с текстами :)

В любом случае, надо рассматривать конкретный ресурс или документ и решать в каждом случае, где ему место, а где не место.

Например, сейчас в разделе "Пользователь" просто стоит логин/пароль, но когда эта функция будет активной, там появятся все ресурсы предназначенные специально для данного пользователя.

**Comment by Sergei-Litvinov** [comment]
Я думаю, что текущая планировка начальной страницы должна относиться к разделу "Читать КУ" или "Исследовать КУ", а собственно "Главная страница" должна содержать полную структуру сайта с неким объяснением нашего подхода к его построению, а также краткой аннотацией ресурсов и возможностей для пользователя, предоставляемых нашим ресурсом.

**Comment by tigran123** [comment]
Дело в том, что в данный момент секции "ГЛАВНАЯ", "ЭКЗАМЕН", "ПОЛЬЗОВАТЕЛЬ" и "НАСТРОЙКИ" являются одним целым, т.е. частью программы "Исследователь Книги Урантии" и выносить их куда-то "за скобки" в отдельный ящик не стоит. Например, при работе в секции "ЭКЗАМЕН" пользователь будет перенаправляться в секцию "ГЛАВНАЯ" для углубления знаний по какому-то вопросу (в случае неправильного ответа на вопрос).  От секции "ПОЛЬЗОВАТЕЛЬ" тоже есть прямая зависимость --- в зависимости от введённого логина/пароля в секции "ГЛАВНАЯ" произойдут изменения --- те или иные модули заметок будут доступны или недоступны. Тоже самое "НАСТРОЙКИ" --- в зависимости от выбранных значений, сама программа "Исследователь Книги Урантии" будет работать определённым образом.

Более того, очевидно, что именно "Исследователь Книги Урантии" является ядром содержащим 99% логики и ценности всей работы и будет смешно его куда "прятать", а статическое содержимое сайта называть "Главной страницей". Остальные материалы я рассматриваю как "более или менее полезный довесок" который можно конечно добавить куда-нибудь в меню, так чтобы и доступно было и не мешало серьёзной работе с текстом.

Сергей, мы создаём нечто уникальное и потенциально очень полезное и, в таких случаях, не надо смотреть "как делают другие". Точнее, смотреть обязательно надо, но иммитировать не стоит. Другие решают более простые задачи и выбирают более простую схему. Будем оставаться самими собой и действовать оптимально в смысле логики и функционала, а не в смысле "конвенции" и "похожести" на других. Иначе получится смешная ситуация как, скажем, на urantia.ru --- там оказывается стоит супер-сложный и интересный алгоритм поиска (я бы весь их "сайт" выкинул и оставил только эту функцию поиска, растянутую на всё окно, со всеми своими опциями) а он спрятан куда-то в маленькое незаметное окошко слева и я, например, 6 лет знал о существовании этого сайта, но ни разу (пока ты мне не указал на интересный вариант поисковика на нём пару дней назад) им не пользовался --- именно из-за всяких "дополнительных материалов" а также очень неудобно представленного текста Книги Урантии (разбитого на архаичные "страницы" --- это же век динозавров!) которые мне показались ничем не привлекательными и, как говорится, "гроша ломаного не стоящими" --- а за ними оказался спрятан чистый алмаз очень умного поисковика у которого действительно стоит кое-чему поучиться.

Другое дело, что если "оставаясь самим собой" мы создаём неудобство кому-то в работе или недостачу какой-то функции --- это совсем другое дело и в каждом конкретном случае должно быть рассмотрено серьёзно и дефект устранён. Какой конкретно функции тебе не хватает? Вот когда она появится в чётко-сформулированной форме, тогда её мы и добавим, и именно туда, куда тебе (и остальным пользователям) будет удобно.

Это только у "корпоративных капиталистов" так поставлено: работа хакера где-то на "заднем плане" (хотя на ней всё держится), а мишура какого-нибудь маркетроида растянута на весь сайт. Но мы здесь все хакеры и сайт тоже для хакеров, так что... ;)

PS. Мы живём на планете, где всё настолько искажено, что даже языком (любым, каким именно --- не важно) становится труднее и труднее пользоваться, ибо значения многих важных слов изменены с целью их опошления и постепенного затуманивания сознания человека ведущего к отказу от этих слов. Например, выше я использовал слово "хакер" в смысле, употребляемом самим хакерами, т.е. "хакер", по _определению_ это технически образованный человек, посвятивший свою жизнь бескорыстному служению ближнему. Настоящий хакер никогда не причинит зла и не будет "взламывать" ничей компьютер (для такого типа людей у нас есть термин "скрипт-кидди" (script kiddie) --- т.е. недоразвитый выскочка, который только "скрипты" научился писать, а уже из себя что-то строит). Хакер конечно может это сделать (и даже захватить сеть целого крупного государства если необходимо) но не станет делать, ибо приобретал знания с целью служения любви, а не причинения вреда. А "корпоративные" внушают людям ложь, что якобы "хакеры" это что-то опасное и что якобы они "пишут вирусы" или ещё какую-нибудь дрянь...  Эта та же самая специально разжигаемая (капиталистами) ненависть, основанная на страхе: капиталисты ещё как боятся хакеров --- ибо не будь сегодня хакеров, то не будет и сколько-нибудь сложных продуктов --- кстати, поэтому капиталисты и оболванивают народ, придумывая такие никчёмные понятия как "user-friendly" и т.д., чтобы "сколько-нибудь сложные" продукты и не нужны были, а "несложные" и профан может создать для другого такого же профана.

**Comment by tigran123** [comment]
Кстати, добавлю один интересный момент: сами работники Фонда Урантии (сайтом которых кое-кто восхищался) считают дизайн своего сайта _ужасным_ (да, они так и сказали в письме мне, именно "ужасным и примитивным") и, увидев первые наброски моего дизайна, очень обрадовались (буквально сказали, что это "ответ на их молитвы" --- я не шучу, честное слово) и заинтересовались, насколько сложно будет прилепить это к их сайту в качестве замены их собственного "ужасного и примитивного" поисковика. Т.е. я хочу сказать, что даже сами создатели тех сайтов, которые на первый взгляд есть искушение "иммитировать" иногда прекрасно понимают, что принятые ими решения на самом деле очень неудобны и хотелось бы иметь что-нибудь посложнее и погибче в техническом смысле.
